tests/vm_test.go

```
@@ -828,6 +828,43 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 			})
 		})
 
+		It("should delete a VM with high TerminationGracePeriod immediately when TerminationGracePeriodSeconds is set to 0", Serial, func() {
+			vm := libvmi.NewVirtualMachine(
+				libvmifact.NewAlpine(libvmi.WithTerminationGracePeriod(1600)),
+				libvmi.WithRunStrategy(v1.RunStrategyAlways))
+
+			By("Creating a VM with a high TerminationGracePeriod")
+			vm, err = virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Create(context.Background(), vm, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting for VMI to start")
+			Eventually(ThisVM(vm), 360*time.Second, 1*time.Second).Should(BeReady())
+
+			vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			pod, err := libpod.GetPodByVirtualMachineInstance(vmi, vmi.Namespace)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Force Deleting the VM")
+			gracePeriod := int64(0)
+			err = virtClient.VirtualMachine(vm.Namespace).Delete(context.Background(), vm.Name, metav1.DeleteOptions{GracePeriodSeconds: &gracePeriod})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Ensuring the VMI reflects the updated TerminationGracePeriodSeconds")
+			Eventually(func(g Gomega) {
+				vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(*vmi.Spec.TerminationGracePeriodSeconds).To(BeEquivalentTo(0))
+			}, 60*time.Second, 5*time.Second).Should(Succeed(), "VMI should have been updated by the patch")
```

I might be overthinking this but is 5 seconds going to be too long to poll the VMI again given grace period of < 5? Do we need to lower the retry time or merge the two Eventually blocks here and just ignore errors looking up the VMI if the VM has also been deleted?

---

i adjusted the poll times to 1 seconds, and also added consistently block to make sure the pod wont be terminated for the requested grace time